signer/src/api/new_block.rs

```diff
@@ -112,34 +162,251 @@ pub async fn new_block_handler(state: State<ApiState<impl Context>>, body: Strin
+///   fullfilling tx.
+/// - `bitcoin_block_height`: The height of the Bitcoin block containing the
+///   fullfilling tx.
+/// - `stacks_chaintip`: Current Stacks blockchain chaintip information for
+///   context on block height and hash.
+///
+/// # Returns
+/// - `Result<WithdrawalUpdate, Error>`: On success, returns a `WithdrawalUpdate` struct
+///   for Emily containing relevant withdrawal information.
+///   In case of a database error, returns an `Error`
+async fn handle_withdrawal_accept(
+    ctx: &impl Context,
+    event: WithdrawalAcceptEvent,
+    stacks_chaintip: &StacksBlock,
+    // TODO (#493): We need the `bitcoin_block_hash` and `bitcoin_block_height`
+    // of the block that included the fulfilling Bitcoin transaction.
+    // After #493 is resolved, this value should be contained in the event itself
+    // and these parameters should be removed.
+    bitcoin_block_hash: String,
+    bitcoin_block_height: u64,
+) -> Result<WithdrawalUpdate, Error> {
+    ctx.get_storage_mut()
+        .write_withdrawal_accept_event(&event)
+        .await?;
+
+    Ok(WithdrawalUpdate {
+        request_id: event.request_id,
+        status: Status::Confirmed,
+        fulfillment: Some(Some(Box::new(Fulfillment {
+            bitcoin_block_hash,
+            bitcoin_block_height,
+            bitcoin_tx_index: event.outpoint.vout,
+            bitcoin_txid: event.outpoint.txid.to_string(),
+            btc_fee: event.fee,
+            stacks_txid: event.txid.to_hex(),
+        }))),
+        status_message: format!("Included in block {}", event.block_id.to_hex()),
+        last_update_block_hash: stacks_chaintip.block_hash.to_hex(),
+        last_update_height: stacks_chaintip.block_height,
+    })
+}
+
+/// Processes a withdrawal creation event, adding new withdrawal records to the
+/// database and preparing the data for Emily.
+///
+/// # Parameters
+/// - `ctx`: Shared application context containing configuration and database access.
+/// - `event`: The withdrawal creation event to be processed.
+/// - `stacks_block_height`: The height of the Stacks block containing the withdrawal tx.
+///
+/// # Returns
+/// - `Result<CreateWithdrawalRequestBody, Error>`: On success, returns a `CreateWithdrawalRequestBody`
+///   with withdrawal information. In case of a database error, returns an `Error`
+async fn handle_withdrawal_create(
+    ctx: &impl Context,
+    event: WithdrawalCreateEvent,
+    stacks_block_height: u64,
+) -> Result<CreateWithdrawalRequestBody, Error> {
+    ctx.get_storage_mut()
+        .write_withdrawal_create_event(&event)
+        .await?;
+
+    Ok(CreateWithdrawalRequestBody {
+        amount: event.amount,
+        parameters: Box::new(WithdrawalParameters { max_fee: event.max_fee }),
+        recipient: event.recipient.to_string(),
+        request_id: event.request_id,
+        stacks_block_hash: event.block_id.to_hex(),
+        stacks_block_height,
```

Comment:
@djordon once the PRs you mentioned are implemented, all sBTC contract events will include the data needed to create updates in Emily. Currently, WithdrawalCreateEvent is the only outlier, as it has the hash but not stacks_block_height. Do you think it’d be worthwhile to request this field in the sbtc-registry contract’s print event for consistency? It’s not a blocker, but it would align with the others. If so, could you point me to the relevant issue, or should I open a new one?

Reply:
Ohh, I don't think we need that in the print event. I think for `WithdrawalCreateEvent` you should use the block height included in the webhook, just like you're doing now.